### ISSUE TITLE
Problem editing saved manual pireps

### ISSUE THREAD

**Issue by jonbeckett** [FatihKoz/DisposableTheme#20]
We're seeing a strange problem when editing saved manual pireps...

![Screenshot (1170)](https://github.com/FatihKoz/DisposableTheme/assets/327583/805a63f1-7cde-4aae-97ec-8f146a2f0054)

Any ideas ? (not very familiar with the framework yet, otherwise I might start tinkering lol)

**Comment by jonbeckett** [comment]
Steps to reproduce - create a manual pirep, and save it - don't submit it. Then go to your reports, and click the edit button next to the pirep.

**Comment by jonbeckett** [comment]
Not sure if this is related - the queries generated to get the airport(s) are wrong - there are no quotes around the ICAO codes in the IN clause.

![Screenshot (1171)](https://github.com/FatihKoz/DisposableTheme/assets/327583/275b663a-a0c0-418d-9823-f48ca4739fc8)

**Comment by jonbeckett** [comment]
The $airport_list coming from phpvms is populated with an array by the edit controller - not a single value - but then your code loops through them... strange.

![Screenshot (1172)](https://github.com/FatihKoz/DisposableTheme/assets/327583/a83d7b91-8a8f-4c17-b40e-85e2552e3db8)

**Comment by FatihKoz** [comment]
Looks like the controller is sending a multidimensional array where we expect to have the airport name/label when editing a pirep.

```
  0 => [  "" => "" ],
  1 => [ $dep_airport_id => $dep_airport_label ],
  2 => [ $arr_airport_id => $arr_airport_label  ],
  3 => [ $alt_airport_id => $alt_airport_label  ],
```

And this generates an error when we want to display the label with `{{ $dpt_airport_label }}` directly. This works with default theme or when we use old `laravelcollective/html` package for forms, but fails with pure html form.

Technically (as the best solution) it needs a controller change to prepare and send the array like below

```php
$airports = [
  $dep_airport_id => $dep_airport_label,
  $arr_airport_id => $arr_airport_label,
  $alt_airport_id => $alt_airport_label,
]
```

Doing this may cause problems with default theme or any theme using `laravelcollective/html` package, so an interim solution will be something like below

```php
@foreach($airport_list as $key => $alist)
    @foreach($alist as $airport_id => $airport_label)
        @if(filled($airport_id))
            <option value="{{ $airport_id }}" @if(!empty($pirep) && $pirep->dpt_airport_id == $airport_id) {{ 'selected' }} @endif>{{ $airport_label }}</option>
         @endif
    @endforeach
@endforeach
```

Looks weird but works :wink:

![image](https://github.com/FatihKoz/DisposableTheme/assets/74361521/03f40570-dc61-44e8-b32b-b10faaee516e)

We need to fix this blade again when the controller gets updated with the removal of `laravelcollective/html` package from phpvms (or default theme) though, but to save the day this is acceptable.

Closed with 11271ceead315501d926b0b405b2c30730d0948f

**Comment by jonbeckett** [comment]
Cheers for looking at this - I was in the office most of the day and wasn't able to really look properly :)  I guessed it was to do with the array structure being returned but didn't want to fiddle with it too much :)
